src/Controller/DatatransResponseController.php

```diff
@@ -31,6 +31,15 @@ class DatatransResponseController {
    */
   public function processSuccessResponse(Request $request, PaymentInterface $payment) {
     try {
+      if (\Drupal::state()->get('datatrans.debug', FALSE)) {
+        if(\Drupal::moduleHandler()->moduleExists('past')) {
+          past_event_save('datatrans', $payment->id(), $request);
```

Comment:
This isn't correct. the second argument should be a static machine name to describe what kind of log entry this is, then a user readable label instead of $request and then some arguments, including data from the payment and the request. Instead of $request, it might be much more useful to log $post_data instead.